docs/examples/using-deno-kv-oauth.md

```diff
@@ -0,0 +1,159 @@
+---
+description: Easily integrate OAuth 2.0 to your Fresh project with Deno KV OAuth.
+---
+
+> ⚠️ Please note that this functionality relies on
+> [Deno KV](https://deno.com/manual/runtime/kv) which is currently marked as
+> **experimental** and is **subject to change**. It is only available when using
+> the `--unstable` flag.
+
+> ⚠️ Please note that Deno KV Oauth is still in **beta** and is **subject to
+> change**. See [the documentation](https://deno.land/x/deno_kv_oauth) for
+> further details.
+
+## Using a pre-configured OAuth 2.0 provider
+
+This example uses GitHub as the OAuth 2.0 provider. However there is a suite of
+[pre-configured providers](https://deno.land/x/deno_kv_oauth#pre-configured-oauth-20-clients)
+available.
+
+1. Register a [new GitHub OAuth](https://github.com/settings/applications/new)
+   application, if you haven't already.
+2. Create your pre-configured OAuth client instance. For reusability the
+   instance is stored in `utils/oauth-client.ts`.
+
+```typescript
+// utils/oauth-client.ts
+import { createGitHubOAuth2Client } from "https://deno.land/x/deno_kv_oauth/mod.ts";
+
+export const oauth2Client = createGitHubOAuth2Client();
+```
+
+3. Using the OAuth 2.0 client instance, insert the authentication flow functions
+   into your authentication routes. In this example, there are dedicated handler
+   routes at `routes/auth/signin.ts`, `routes/auth/signout.ts` and
+   `routes/auth/callback.ts`. Please ensure that the `callback` handler matches
+   the Authorization callback URL in the configured OAuth application!
+
+```typescript
+// routes/auth/signin.ts
```

Comment:
```suggestion
// routes/signin.ts
```
We want the user to do the bare minimum to get OAuth working. Flattening the route structure means one less folder that the user has to create.